pkg/i2gw/providers/gce/gce_extensions_test.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gce
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	backendconfigv1 "k8s.io/ingress-gce/pkg/apis/backendconfig/v1"
+)
+
+func TestGetBackendConfigMapping(t *testing.T) {
+	t.Parallel()
+	testNamespace := "test-namespace"
+
+	testServiceName := "test-service"
+	testBeConfigName1 := "backendconfig-1"
+	testBeConfigName2 := "backendconfig-2"
+	backendConfigs := map[types.NamespacedName]*backendconfigv1.BackendConfig{
+		{Namespace: testNamespace, Name: testBeConfigName1}: {},
+		{Namespace: testNamespace, Name: testBeConfigName2}: {},
+	}
+	expectedServices := serviceNames{
+		{Namespace: testNamespace, Name: testServiceName},
+	}
+
+	testCases := []struct {
+		desc             string
+		services         map[types.NamespacedName]*apiv1.Service
+		expectedServices serviceNames
+	}{
+		{
+			desc: "Specify BackendConfig with cloud.google.com/backend-config annotation, using the same BackendConfig for all ports",
+			services: map[types.NamespacedName]*apiv1.Service{
+				{Namespace: testNamespace, Name: testServiceName}: {
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      testServiceName,
+						Namespace: testNamespace,
+						Annotations: map[string]string{
+							backendConfigKey: `{"default":"backendconfig-1"}`,
+						},
+					},
+				},
+			},
+			expectedServices: expectedServices,
+		},
+		{
+			desc: "Specify BackendConfig with beta.cloud.google.com/backend-config annotation, using the same BackendConfig for all ports",
+			services: map[types.NamespacedName]*apiv1.Service{
+				{Namespace: testNamespace, Name: testServiceName}: {
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      testServiceName,
+						Namespace: testNamespace,
+						Annotations: map[string]string{
+							betaBackendConfigKey: `{"default":"backendconfig-1"}`,
```

Comment:
Nit: Should have a test with both beta and v1 annotations specified

Reply:
Added the test cases, thanks!